packages/core/src/interpreter.ts

```diff
@@ -197,24 +193,15 @@ export class Interpreter<
     });
   }
   public get initialState(): State<TContext, TEvent> {
-    if (!IS_PRODUCTION) {
-      warn(
-        this.initialized,
-        // tslint:disable-next-line:max-line-length
-        `Attempted to read initial state from uninitialized service '${this.id}'. Make sure the service is started first.`
-      );
-    }
-
-    return (
-      this._initialState ||
-      withServiceScope(this, () => this.machine.initialState)
```

Comment:
usage of `withServiceScope` was unnecessary - reading an `initialState` can't escape a "single machine boundary" and it's pure, so it doesn't affect spawn actions etc

Reply:
I think there were issues with services being invoked at the initial state (e.g., in an initial `assign()` action) but those should be solved another way, so this is OK